gen_metadata.py

```diff
@@ -910,37 +924,74 @@ def shrink_bbox(mask):
     return cmin, rmin, cmax, rmax
 
 
-def gen_metadata_safe(file_path):
+def gen_metadata_safe(file_path, device=None, maskfname=None, visfname=None):
     """
     Deals with erroneous metadata generation errors.
     """
     try:
-        return gen_metadata(file_path)
+        return gen_metadata(file_path, device=device, maskfname=maskfname, visfname=visfname)
     except Exception as e:
         print(f'{file_path}: Errored out ({e})')
         return {file_path: {'errored': True}}
 
 
+def argument_parser():
+    parser = argparse.ArgumentParser(description='Generate metadata for one or more fish images.')
+    parser.add_argument('file_or_directory',
+                        help='Path to a fish image or a directory of multiple fish images. '
+                             'When one file is passed the JSON metadata is printed to the terminal.')
+    parser.add_argument('limit', type=int, nargs='?',
+                        help='Limit the number of images processed from a directory')
+    parser.add_argument('--outfname',
+                        help='Output filename to use for JSON metadata (disables printing to terminal).')
```

Comment:
Shouldn't this say as well _Only supported when processing a single image file_, to be consistent with the other options with this constraint?